src/Core/Content/Product/SalesChannel/Listing/ProductListingRoute.php

```diff
@@ -166,4 +166,41 @@ protected function doSearch(Criteria $criteria, SalesChannelContext $context): E
 
         return $result;
     }
+
+    protected function isRouteSupported(Request $request): bool
+    {
+        // Findologic should never trigger on home page, even if there are categories that would allow it.
+        if ($this->isHomePage($request)) {
+            return false;
+        }
+
+        // In case requests come from the home page, Findologic should not trigger on those.
+        if ($this->isRequestFromHomePage($request)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function isHomePage(Request $request): bool
+    {
+        return $request->getPathInfo() === '/';
```

Comment:
Should this check be moved to the method below and `return` if this is the case?

Reply:
I want these two scenarios to be separate as their origins are different. This one tests for the homepage itself, while the other one checks for any route having the referer of the homepage.